src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3565,32 +3556,13 @@ private bool IsPossibleParameter()
             }
         }
 
-        private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter, SyntaxListBuilder<AttributeListSyntax> attributes, SyntaxListBuilder modifiers)
+        private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
         {
             if (parameter == null)
             {
                 return false;
             }
 
-            // cannot reuse parameter if it had attributes.
-            //
-            // TODO(cyrusn): Why?  We can reuse other constructs if they have attributes.
-            if (attributes.Count != 0 || parameter.AttributeLists.Count != 0)
-            {
-                return false;
-            }
-
-            // cannot reuse parameter if it had modifiers.
-            if ((modifiers != null && modifiers.Count != 0) || parameter.Modifiers.Count != 0)
-            {
-                return false;
```

Comment:
>return false; [](start = 16, length = 13)

this wasn't needed?

Reply:
Not anymore.  Hree's the idea:

when incrementally consuming a node from the *previous* tree, we cann't use the node if we've parsed out any pieces from the *next* text file. If we did that, then we'd have an inconsistency where we had an old node, but some new tokens/nodes that we didn't know what to do with.  So we used to say "if we have parsed out any attributes/modifiers, then we can't incrementally use an old node".

This is clearly not great for incrementla purposes, and is one of the reasons we do *not* want parsing functions to take in state.   Now, things are just simpler, the loop that is parsing out parameters just tries to get parameters.  We can consume parameters incrementally in any circumstance since now we only try to get the attributes/modifiers if we did *not* incrementally consume anything.

Reply:
@jcouv did https://github.com/dotnet/roslyn/pull/38935#discussion_r329709632 make sense?

Reply:
Yup. Thanks!